### PR TITLE
fix(lsp.reference): single reference direct jump

### PR DESCRIPTION
**Pull request by jamestrew**
follow up to https://github.com/nvim-telescope/telescope.nvim/pull/3099
> clean up `list_to_jump` to work mostly with `vim.lsp.util.locations_to_items` output from the get-go

and closes https://github.com/nvim-telescope/telescope.nvim/issues/3118

### CODE REVIEW

**Comment by Conni2461** [comment]
i am going to review this tomorrow

File: lua/telescope/builtin/__lsp.lua [hunk 1]
```diff
@@ -97,51 +97,74 @@ lsp.outgoing_calls = function(opts)
...
+---| "textDocument/definition"
+---| "textDocument/typeDefinition"
+---| "textDocument/implementation"
+
+-- luacheck: push ignore
+---@type { [telescope.lsp.list_or_jump_action]: fun(items: vim.lsp.util.locations_to_items.ret[], opts: table): vim.lsp.util.locations_to_items.ret[] }
+-- luacheck: pop
 local action_handlers = {
-  ["textDocument/references"] = function(results, items, opts)
+  ["textDocument/references"] = function(items, opts)
```
Comment (Conni2461) [inline lua/telescope/builtin/__lsp.lua:134]: it doesnt make sense to have an "action_handers" table here, it has one entry, why would we want to construct a table for that.

we could just do this in `apply_action_handler`
```lua
if action == "textDocument/references" then
  -- stuff from function
end
return locations, items
```

Comment (Conni2461) [reply lua/telescope/builtin/__lsp.lua:134]: its a fine idea for an abstraction but we dont need it here, we have 1 entry, we can still introduce an abstraction if we have like 3, 4, 5 entries

File: lua/telescope/builtin/__lsp.lua [hunk 2]
```diff
@@ -97,51 +97,74 @@ lsp.outgoing_calls = function(opts)
...
----@param action string
+---@param action telescope.lsp.list_or_jump_action
 ---@param title string prompt title
 ---@param params lsp.TextDocumentPositionParams
 ---@param opts table
 local function list_or_jump(action, title, params, opts)
+  opts.reuse_win = vim.F.if_nil(opts.reuse_win, false)
+
+  local curr_filepath = vim.api.nvim_buf_get_name(opts.bufnr)
+
```
Comment (Conni2461) [inline lua/telescope/builtin/__lsp.lua:167]: please remove empty line

**Comment by Conni2461** [comment]
works for both 0.9 and 0.10, can be merged after two nits are addressed, thanks. i like that solution more than #3120

**Comment by jamestrew** [comment]
thanks for reviewing!

**Comment by jamestrew** [comment]
@lawrenceho that's unfortunate. I'll see what we can do to remedy this.

**Comment by lawrenceho** [comment]
Thank you @jamestrew. By the way, I can create a PR for this by adopting the same approach used in `vim.uri_from_bufnr`.

**Pull request merged**